sentry_sdk/tracing_utils.py

```diff
@@ -413,6 +418,85 @@ def has_custom_measurements_enabled():
     return bool(options and options["_experiments"].get("custom_measurements"))
 
 
+class Baggage(object):
+    __slots__ = ("sentry_items", "third_party_items", "mutable")
+
+    SENTRY_PREFIX = "sentry-"
+    SENTRY_PREFIX_REGEX = re.compile("^sentry-")
+
+    # DynamicSamplingContext
+    DSC_KEYS = [
+        "trace_id",
+        "public_key",
+        "sample_rate",
+        "release",
+        "environment",
+        "transaction",
+        "user_id",
+        "user_segment",
+    ]
+
+    def __init__(
+        self,
+        sentry_items,  # type: Dict[str, str]
+        third_party_items="",  # type: str
+        mutable=True,  # type: bool
+    ):
+        self.sentry_items = sentry_items
+        self.third_party_items = third_party_items
+        self.mutable = mutable
+
+    @classmethod
+    def from_incoming_header(cls, header):
+        # type: (Optional[str]) -> Baggage
+        """
+        freeze if incoming header already has sentry baggage
+        """
+        sentry_items = {}
+        third_party_items = ""
+        mutable = True
+
+        if header:
+            for item in header.split(","):
+                item = item.strip()
+                key, val = item.split("=")
+                if Baggage.SENTRY_PREFIX_REGEX.match(key):
+                    baggage_key = unquote(key.split("-")[1])
+                    sentry_items[baggage_key] = unquote(val)
+                    mutable = False
+                else:
+                    third_party_items += val
+
+        return Baggage(sentry_items, third_party_items, mutable)
+
+    def freeze(self):
+        # type: () -> None
+        self.mutable = False
+
+    def dynamic_sampling_context(self):
+        # type: () -> Dict[str, str]
+        header = {}
+
+        for key in Baggage.DSC_KEYS:
+            if self.sentry_items.get(key):
```

Comment:
sentry_items is accessed twice. would saving into local variable and then do the if against that var and saving that var in `header` make things more performant?

Reply:
it's a variable on the object anyway, don't think that's necessary

Reply:
oh sorry you meant the hash access, yea will do